[
    {
        "content": "<p>I'm trying to diagnose some weird failure (<code>roc run</code> building the host but then exiting with code 9?) and not having a great time of it. Is there a way to get Roc to give verbose logging output to help figure out what's going on here?</p>\n<p>(This only happens on CI, FWIW! If it were local I'd have a lot more tools.) <a href=\"https://github.com/roc-lang/rbt/actions/runs/3044411941/jobs/4904773008\">https://github.com/roc-lang/rbt/actions/runs/3044411941/jobs/4904773008</a></p>",
        "id": 298559897,
        "sender_full_name": "Brian Hicks",
        "timestamp": 1663067204
    },
    {
        "content": "<p><span class=\"user-mention\" data-user-id=\"454654\">@Ayaz Hafiz</span> may be able to help you here, since the failure only happens on x86_64 macos.</p>",
        "id": 298562896,
        "sender_full_name": "Anton",
        "timestamp": 1663068443
    },
    {
        "content": "<p>well, it seems like it maybe happens on Linux too. Iâ€™ve got the integration tests disabled for that target right now</p>",
        "id": 298573574,
        "sender_full_name": "Brian Hicks",
        "timestamp": 1663072474
    },
    {
        "content": "<p>alright, I'll give it a try</p>",
        "id": 298573919,
        "sender_full_name": "Anton",
        "timestamp": 1663072589
    },
    {
        "content": "<p>On linux I get:</p>\n<div class=\"codehilite\"><pre><span></span><code>---- examples::test_hello_world stdout ----\nOk(\n    Output {\n        status: ExitStatus(\n            unix_wait_status(\n                25856,\n            ),\n        ),\n        stdout: &quot;ðŸ”¨ Rebuilding host...\\n&quot;,\n        stderr: &quot;thread &#39;&lt;unnamed&gt;&#39; panicked at &#39;Failed to rebuild src/main.rs - stderr of the `cargo build` command was:\\nerror: no bin target named `host`\\n&#39;, crates/compiler/build/src/link.rs:1310:27\\nnote: run with `RUST_BACKTRACE=1` environment variable to display a backtrace\\nthread &#39;main&#39; panicked at &#39;called `Result::unwrap()` on an `Err` value: Any { .. }&#39;, crates/cli/src/build.rs:324:46\\n&quot;,\n    },\n)\nthread &#39;examples::test_hello_world&#39; panicked at &#39;assertion failed: false&#39;, tests/examples.rs:25:9\nnote: run with `RUST_BACKTRACE=1` environment variable to display a backtrace\n</code></pre></div>\n<p>This may be expected, as is mentioned above the test:</p>\n<div class=\"codehilite\"><pre><span></span><code>// example tests are disabled for Unix right now since there&#39;s a bug preventing\n// them from running[...]\n</code></pre></div>\n<p>So probably best to continue on MacOS x86_64</p>",
        "id": 298576906,
        "sender_full_name": "Anton",
        "timestamp": 1663073303
    },
    {
        "content": "<p>That Linux issue is due to trying to use the surgical linker but not being set up for it. You probably will get the real issue if you run with <code>--linker=legacy</code></p>",
        "id": 298595806,
        "sender_full_name": "Brendan Hansknecht",
        "timestamp": 1663078796
    },
    {
        "content": "<p>I think I'll just dig out my x86 mac and try it locally. Gonna have some serious dust to blow off though. <span aria-label=\"laughing\" class=\"emoji emoji-1f606\" role=\"img\" title=\"laughing\">:laughing:</span></p>",
        "id": 298596039,
        "sender_full_name": "Brian Hicks",
        "timestamp": 1663078851
    },
    {
        "content": "<p>welp, dust blown off, upgraded to macOS 12 dot whatever, and I can't reproduce the bug. <span aria-label=\"rolling eyes\" class=\"emoji emoji-1f644\" role=\"img\" title=\"rolling eyes\">:rolling_eyes:</span> Soooooo once again: are there any \"BE VERY VERBOSE\" flags or environment variables I can throw in here to get more info out of CI?</p>",
        "id": 298634901,
        "sender_full_name": "Brian Hicks",
        "timestamp": 1663090854
    },
    {
        "content": "<p>also if anyone knows what exit status 9 would mean, that'd answer some questions!</p>",
        "id": 298636919,
        "sender_full_name": "Brian Hicks",
        "timestamp": 1663091604
    },
    {
        "content": "<p>(specifically the question: what does exit status 9 mean?)</p>",
        "id": 298636960,
        "sender_full_name": "Brian Hicks",
        "timestamp": 1663091618
    },
    {
        "content": "<p>If the exit codes are the same as on linux; it means that the system killed your process, a common cause of that seems to be running out of memory.</p>",
        "id": 298639484,
        "sender_full_name": "Anton",
        "timestamp": 1663092410
    },
    {
        "content": "<p>I don't think looking at verbose compiler output will help here, my first step would be to download the executable that ci used for the test and see what it does on your machine.</p>",
        "id": 298639737,
        "sender_full_name": "Anton",
        "timestamp": 1663092491
    },
    {
        "content": "<p>Example on how to upload the executable: <a href=\"https://github.com/roc-lang/roc/blob/4644ce40e3f492af2df6515ab7ae290f8221a243/.github/workflows/nix_linux_x86_64.yml\">https://github.com/roc-lang/roc/blob/4644ce40e3f492af2df6515ab7ae290f8221a243/.github/workflows/nix_linux_x86_64.yml</a></p>",
        "id": 298639856,
        "sender_full_name": "Anton",
        "timestamp": 1663092542
    },
    {
        "content": "<p>I'd also inspect memory usage when just running the test on your own machine</p>",
        "id": 298640110,
        "sender_full_name": "Anton",
        "timestamp": 1663092613
    },
    {
        "content": "<p>OK, here's something curious: it works if I build a binary and then run the binary instead of just doing <code>roc run</code> <span aria-label=\"thinking\" class=\"emoji emoji-1f914\" role=\"img\" title=\"thinking\">:thinking:</span></p>",
        "id": 298651639,
        "sender_full_name": "Brian Hicks",
        "timestamp": 1663096789
    },
    {
        "content": "<p>I've also seen that behavior in a different error <a href=\"https://github.com/JanCVanB/roc-plotters/pull/5#issuecomment-1232945686\">here</a> but I have not yet found the cause.</p>",
        "id": 298715528,
        "sender_full_name": "Anton",
        "timestamp": 1663139933
    }
]